lib/galaxy/datatypes/molecules.py

```diff
@@ -19,24 +18,28 @@
 from galaxy.datatypes.tabular import Tabular
 from galaxy.datatypes.util.generic_util import count_special_lines
 from galaxy.datatypes.xml import GenericXml
-from galaxy.util import unicodify
+from galaxy.util import (
+    commands,
+    unicodify
+)
 
 log = logging.getLogger(__name__)
 
 
 def count_lines(filename, non_empty=False):
     """
-        counting the number of lines from the 'filename' file
+    counting the number of lines from the 'filename' file
     """
+    if non_empty:
+        cmd = ['grep', '-cve', r'^\s*$', filename]
+    else:
+        cmd = ['wc', '-l', filename]
     try:
-        if non_empty:
-            out = subprocess.Popen(['grep', '-cve', r'^\s*$', filename], stdout=subprocess.PIPE)
-        else:
-            out = subprocess.Popen(['wc', '-l', filename], stdout=subprocess.PIPE)
-        return int(out.communicate()[0].split()[0])
-    except Exception:
-        pass
-    return 0
+        out = commands.execute(cmd)
+    except commands.CommandLineException as e:
+        log.error(str(e))
```

Comment:
```suggestion
        log.error(unicodify(e))
```

Reply:
Is that necessary ? `CommandLineException` already ensures that user-input is in unicode

Reply:
Mainly for consistency, since we did a pass recently adding `unicodify()` around all exceptions in `log.method("Message %s" % e)` calls. Also I don't think we make sure that the command part of the `CommandLineException` message is Unicode.

Reply:
Done, here and in all other places changed by this PR.